src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const copy = [...this];
+    let res = '';
+    let sep = separator + '';
+
+    if (typeof separator === 'number') {
+      return;
+    }
```

Comment:
No need to check type if you already converted it
```suggestion
    const separatorString = String(separator);
```